lib/rubocop/cop/rspec/empty_example_group.rb

```diff
@@ -68,13 +68,20 @@ class EmptyExampleGroup < Cop
         PATTERN
 
         def on_block(node)
-          return unless example_group?(node) && !contains_example?(node)
+          return unless example_group?(node)
+          return if node.children.any? do |c|
+            c && should_contain_example?(c) && contains_example?(c)
```

Comment:
You can use `each_child_node` and you won't have non-parser nodes yield, so you can skip the `c &&` part